app/controllers/config_reports_controller.rb

```diff
@@ -1,10 +1,20 @@
 class ConfigReportsController < ApplicationController
   include Foreman::Controller::AutoCompleteSearch
+  include Foreman::Controller::CsvResponder
 
   before_action :setup_search_options, :only => :index
 
   def index
-    @config_reports = resource_base_search_and_page(:host)
+    respond_to do |format|
+      format.html do
+        @config_reports = resource_base_search_and_page(:host)
+        render :index
+      end
+      format.csv do
+        @config_reports = resource_base_with_search.preload(:host)
```

Comment:
@tbrisker this still means that if there are any customization on the controller (e.g. logic to define csv_columns) they will not be covered by a test (not the actual SQL query from the server etc). or am I missing something?